blueprints/ort-rewrite-unix-style.md

```diff
@@ -0,0 +1,210 @@
+
+### Security Impact
+None.
+
+### Upgrade Impact
+None. The rewrite will provide a compatibility script, so existing tools and runs continue to work as before.
+
+### Operations Impact
+Operators should learn the new real ORT interface, to better operate Traffic Control. However, it's not strictly required, they can continue to use the compatibility script for the immediate future.
+
+### Developer Impact
+Developers on ORT will have to learn the new codebase and write in Go.
+
+The new code should not be logically larger or more complex than the existing ORT. The Go language is vertically verbose, but this project does not significnatlly increase the actual logic, or add new logical features.
+
+## Alternatives
+There are innumerable ways ORT could be rewritten. Some possibilities:
+- Transliterated in-place to another language.
+    - Disadvantages: Monolithic, more difficult to develop, more difficult to understand each component and "thing" done, more difficult to understand all things done.
+    - Advantages: faster, depending on the developer. Execution may be faster (avoids Inter-Process Communication).
+- Refactored into the UNIX Philosophy with the current Perl code.
+    - Disadvantages: still Perl, still unsafe, not compiled, archaic, difficult to write and read.
+    - Advantages: faster, depending on the developer.
+- Written as a Service.
+    - Disadvantages: another single-point-of-failure, another running app Operators have to be aware of and maintain and monitor, uses constant resources like CPU and memory on every Cache, if it communicates with Traffic Ops it potentially creates a undirected graph of the TC network communication which is much more difficult to comprehend and ensure correctness.
+    - More power, can receive requests from Traffic Ops, can run irrespective of CRON or operator deployment.
+- Removed, changed to a server orchestration tool like Ansible or Puppet.
+    - Disadvantages: radical change, makes TC dependent on an orchestration tool and platform, tool will require nearly as much code to do the same thing in its own instruction language, requires operators to learn a particular and very large and complex third-party tool, requires even small CDN Operators to deploy a large orchestration tool they don't otherwise need to learn or deploy, requires CDN operators with a different orchestration tool to deploy and learn multiple large redundant systems.
+    - Advantages: Simpler, an orchestration config language is (hopefully) simpler than a Turing-Complete app. Easier for Operators to learn, if they're already using the orchestrator everywhere else.
+
+## Dependencies
+None anticipated. Small Go library dependencies may be needed, which will be considered as they arise. Any dependencies should be compile-time.
```

Comment:
Yeah, it will definitely have compile-time dependencies unless you plan to write an HTTP client in whatever language you use. At a minimum, it will depend on the standard library. All that should be fine, though. It's hard to guess which dependencies you'll need from the get-go.